### DISCUSSION TITLE
Date and time is not parsing after creating custom NER Model

### DISCUSSION THREAD

**Discussion by sushamasivananthan** [explosion/spaCy#8001] [Help: Coding & Implementations] [answered]
Hi Team,
I'm trying to create a custom NER model to parse informations like Invoice number ,date, time and amount from invoices.

I have used around 50K+ records to train the model, but after training the model can detect the invoice number and tax id etc. but it is unable to detect date and time if there is even a slight change in date format(for example dd/mm/yy instead of dd/mm/yyyy).
Is there any option to combine pre-trained spacy model with my custom NER model. If so what code change should I do?

```
Environment:

spacy  : 3.0.6
OS: linux 
Python: 3.8.5
```

**### I have given my sample data set below.**

`NEW_TRAIN=[('UID:1234 17:28:35 29.2.20 ', {'entities': [(0, 8, 'UID'), (9, 17, 'time'), (18, 25, 'date')]})]`

**### My code looks like this:**

```
import pandas as pd
from tqdm import tqdm
import spacy
from spacy.tokens import DocBin

nlp = spacy.load("en_core_web_sm") # load a new spacy model
db = DocBin() # create a DocBin object

for text, annot in tqdm(NEW_TRAIN):
    #print(text)
    #print(annot)
#for text, annot in tqdm(TRAIN_DATA): # data in previous format
    doc = nlp.make_doc(text) # create doc object from text
    ents = []
    for start, end, label in annot["entities"]: # add character indexes
        #print(start)
        #print(end)
        #print(label)
        span = doc.char_span(start, end, label=label, alignment_mode="contract")
        #print(span)
        if span is None:
            
            print("Skipping entity")
        else:
            ents.append(span)
    doc.ents = ents # label the text with the ents
    db.add(doc)

db.to_disk("./train.spacy")
```

Also, I have created the config file from this link [https://spacy.io/usage/training] where I have mentioned language as **german** and component as **ner**

**### Model training code looks like this**

`python3 -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./train.spacy`

please help.

**Comment by svlandeg** [comment, marked as answer]
Hi!

This sounds like it's the ["catastrophic forgetting"](https://explosion.ai/blog/pseudo-rehearsal-catastrophic-forgetting) problem. In a nutshell, it means that a machine learning model will start "forgetting" things it could predict correctly before, after feeding (lots of) new data to it. The internal weights of the model will adjust themselves to the new examples you're feeding in. As it doesn't get penalized for "forgetting" the old ones, this may happen.

What we recommend is that you feed in typical "old" data in between your "new" annotations, so that the ML models keeps performing well for both types. If you don't have access to gold-standard annotations, you could also generate a bunch of predictions with your original/old model, and feed in those predictions as "gold standard". At least that will ensure that the model keeps making those same predictions (more or less).

The other option would be to have two NER models in the pipeline. It depends on your use-case whether that's viable for you or not. If the entities are entirely different, with different types, this might be an easy fix for your problem though: you just include your new NER model, and you ["source"](https://spacy.io/usage/processing-pipelines#sourced-components) the original NER model from e.g. `en_core_web_sm`.

What you need to be aware of though, is that the second NER model in the pipeline won't be able to overwrite entities that were set by the first. But if your entity sets are largely disjoint, this might not be a problem.
